File: backend/src/main/java/com/example/backend/model/Application.java
package com.example.backend.model;

import com.example.backend.model.enums.ApplicationStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "applications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Application {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @Enumerated(EnumType.STRING)
    private ApplicationStatus status;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false)
    private String personalNumber;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String phoneNumber;

}

File: frontend/src/types/Application.ts
export type ApplicationStatus =
  | 'PENDING'
  | 'APPROVED'
  | 'DISAPPROVED'
  | 'CANCELED'

export interface Application {
  id: string
  createdAt: string
  status: ApplicationStatus
  firstName: string
  lastName: string
  personalNumber: string
  email: string
  phoneNumber: string
  updatedAt: string
}
